pkg/ostree/ostree.go

```diff
@@ -218,6 +210,33 @@ func ResolveRef(location, ref string, consumerCerts bool, subs *rhsm.Subscriptio
 	return checksum, nil
 }
 
+// ResolveRef is deprecated, use ResolveRefMTLS.
+func ResolveRef(location, ref string, consumerCerts bool, subs *rhsm.Subscriptions, ca *string) (string, error) {
```

Comment:
Just break the API.  The mess of keeping deprecated functions around isn't worth the effort.

Reply:
How about I do a followup PR once this is integrated into composer? Because if I remove this function tests immediately breaks, I feel like I would make my life harder by straight-up breaking it.

Reply:
Which tests break?  The integration ones with composer?  Those are non-required for a reason.  They're meant as a warning to the PR creator that they will have to change osbuild-composer when updating.

> How about I do a followup PR once this is integrated into composer?

I'd rather have one clean PR than two messy ones.  The way it is now it's hard to review (though not too hard).  I suspect that a single PR that makes the appropriate change would be easier for me to wrap my head around, rather than try to imagine what it will look like _eventually_.